modin/experimental/core/execution/native/implementations/omnisci_on_native/interchange/dataframe_protocol/column.py

```diff
@@ -196,13 +196,22 @@ def describe_categorical(self) -> Dict[str, Any]:
                 )
             col = col.combine_chunks()
 
+        from .dataframe import OmnisciOnNativeDataframe
+
         col = col.chunks[0]
-        mapping = dict(enumerate(col.dictionary.tolist()))
+        cat_frame = OmnisciOnNativeDataframe.from_arrow(
```

Comment:
`modin.pandas.utils.from_pandas`?

Reply:
low-level `.from_arrow` is preferred as it's much faster because of zero-copying. Check [this comment](https://github.com/modin-project/modin/pull/4763#discussion_r938928526)

Reply:
sorry, I meant `modin.pandas.utils.from_arrow`. see the related discussion here https://github.com/modin-project/modin/pull/4763#discussion_r944259286

Reply:
current implementation feels more straightforward compared to `modin_obj._query_compiler._modin_frame`. Plus it is resilient to changing engine on the fly :)

Reply:
As to OmnisciOnNative, we can leave as is but for pandas on something I would prefer to use type(...).from_pandas. See the discussion above https://github.com/modin-project/modin/pull/4763#discussion_r944633963